test/PersistentOrderedMap.prop.test.mo

```diff
@@ -0,0 +1,215 @@
+// @testmode wasi
+
+import Map "../src/PersistentOrderedMap";
+import Nat "../src/Nat";
+import Iter "../src/Iter";
+import Debug "../src/Debug";
+import Array "../src/Array";
+
+import Deque "../src/Deque";
+import Buffer "../src/Buffer";
+
+import Suite "mo:matchers/Suite";
+import T "mo:matchers/Testable";
+import M "mo:matchers/Matchers";
+
+import Random2 "mo:base/Random";
+
+let { run; test; suite } = Suite;
+
+class MapMatcher(expected : Map.Map<Nat, Text>) : M.Matcher<Map.Map<Nat, Text>> {
+  public func describeMismatch(actual : Map.Map<Nat, Text>, _description : M.Description) {
+    Debug.print(debug_show (Iter.toArray(Map.entries(actual))) # " should be " # debug_show (Iter.toArray(Map.entries(expected))))
+  };
+
+  public func matches(actual : Map.Map<Nat, Text>) : Bool {
+    Iter.toArray(Map.entries(actual)) == Iter.toArray(Map.entries(expected))
+  }
+};
+
+object Random {
+  var number = 4711;
+  public func next() : Nat {
+    number := (15485863 * number + 5) % 15485867;
+    number
+  };
+
+  public func nextNat(range: (Nat, Nat)): Nat {
+    let n = next();
+    let v = n % (range.1 - range.0 + 1) + range.0;
+    v
+  };
+
+  public func nextEntries(range: (Nat, Nat), size: Nat): [(Nat, Text)] {
+    Array.tabulate<(Nat, Text)>(size, func(_ix) { 
+      let key = nextNat(range); (key, debug_show(key)) } )
+  }
+};
+
+let natMap = Map.MapOps<Nat>(Nat.compare);
+
+func mapGen(samples_number: Nat, size: Nat, range: (Nat, Nat)): Iter.Iter<Map.Map<Nat, Text>> {
+  object {
+    var n = 0;
+    public func next(): ?Map.Map<Nat, Text>  {
+      n += 1;
+      if (n > samples_number) { 
+        null 
+      } else {
+        ?natMap.fromIter<Text>(Random.nextEntries(range, size).vals())
+      }
+    }
+  }
+};
+
+
+func run_all_props(range: (Nat, Nat), size: Nat, map_samples: Nat, query_samples: Nat) {
+  func prop(name: Text, prop: Map.Map<Nat, Text> -> Bool): Suite.Suite {
```

Comment:
Please avoid name shadowing of a function `prop` and its argument `prop`.

Reply:
fixed